### PR TITLE
fix api, reworked firestore initialization, and improved our yarn scripts

### LINKED ISSUE #241: Fix queries after db refactor
# Requirements

- [ ] Currently, there's an error when you try to run a query after we refactored the db initializer. Fix it!

### LINKED ISSUE #242: Remove husky
# Requirements

- [ ] Remove husky as we no longer use it. We can always add it in later!

### LINKED ISSUE #243: Productionize our database initializer
# Requirements

- [ ] Let's improve the database initializer methods to be more easy to read
- [ ] Add resilience
- [ ] Make our database instance a singleton
- [ ] Improve how we import credentials for firebase (we should use env variables instead of a service account key file)

### LINKED ISSUE #244: Set up our seed data generator to run when we're using the emulator
# Requirements

- [ ] When the app runs against an emulator, we should load up some seed data
- [ ] Make sure it doesn't seed duplicate data. (don't run more than once)
- [ ] If we can automate it, awesome!

### LINKED ISSUE #245: Let's improve our readme!
# Requirements

- [ ] Let's make our readme updated with our latest changes
   - [ ] Remove husky
   - [ ] Add details around local database setup
   - [ ] Explain how to debug code. 
   - [ ] Improve sections structuring

### PR DESCRIPTION
**Pull request by ermish**
Resolves #241 
Resolves #242 
Resolves #243 
Resolves #244 
Resolves #245 

# What is this PR?
This fixed our API to run queries again and has major improvements to how we connect to our database

# What changed

- [x] Improved our readme! 📓 
- [x] Fixed firestore connection 🙌 
- [x] Solved a bug where the ports where overlapping in firebase emulator UI and our server 🐛 
- [x] moved some files around for consistency
- [x] Reworked firebase connections setup to be more reliable, use a proper singleton firebase instance 🤝 
- [x] Improved our env variables structuring for firebase and determining if we're running against a production database or an emulator.
- [x] Implemented seeding functionality to run when a new emulator is started! woot. It's automatic! (and it only runs once so we don't have a bunch of duplicate data) 🌱 
- [x] removed logs from git
- [x] upgraded our ts config to use the new module loading system! 🎉 
- [ ] Removed husky 🐕 

# Testing instructions
- Start your firebase emulator using `yarn dev:startlocaldb` and then run `yarn dev` and validate there's seed data and you can query it

### CODE REVIEW

**Comment by theyokohamalife** [comment]
@ermish Here's an example query and response from this PR after grabbing the auto-generated id for Facility from the emulator. Including it in the testing section can help speed up the review process. It also gives us a quick view of which fields still need to be addressed in later PRs.

# Query
```
query Facility($facilityId: ID!) {
  facility(id: $facilityId) {
    nameEn
    nameJa
    contact {
      id
      email
      phone
      website
      mapsLink
      address {
        id
        postalCode
        prefectureEn
        cityEn
        addressLine1En
        addressLine2En
        prefectureJa
        cityJa
        addressLine1Ja
        addressLine2Ja
      }
    }
    id
    healthcareProfessionalIds
    healthcareProfessionals {
      id
      names {
        lastName
        firstName
        middleName
        locale
      }
      degrees {
        nameJa
        nameEn
        abbreviation
      }
      spokenLanguages {
        iso639_3
        nameJa
        nameEn
        nameNative
      }
      specialties {
        names {
          name
          locale
        }
      }
      acceptedInsurance
      isDeleted
    }
    isDeleted
  }
}
```

# Response
```
{
  "data": {
    "facility": {
      "nameEn": "Zoo",
      "nameJa": "動物園",
      "contact": {
        "id": null,
        "email": "zoo@test.com",
        "phone": "08000000000",
        "website": "https://zoo.test.com",
        "mapsLink": "",
        "address": {
          "id": null,
          "postalCode": "100-0000",
          "prefectureEn": "Tokyo",
          "cityEn": "Taito",
          "addressLine1En": "1-1-1",
          "addressLine2En": "Ueno",
          "prefectureJa": null,
          "cityJa": null,
          "addressLine1Ja": null,
          "addressLine2Ja": null
        }
      },
      "id": null,
      "healthcareProfessionalIds": [
        "0R7QSKSNZGeMa2dJlvIl"
      ],
      "healthcareProfessionals": null,
      "isDeleted": null
    }
  }
}
```

**Comment by theyokohamalife** [comment]
@ermish Everything LGTM. I would just ask that we keep Husky for now, and I'm really sorry for the trouble! 🙇🏾‍♀️

File: src/firebaseDb.ts [hunk 1]
```diff
@@ -31,6 +31,8 @@ const testFirestoreIsInitialized = async (newDbInstance: Firestore) => {
     }
 }
 
+// This is to prevent race conditions where parallel calls/tests/etc 
+// try to initialize the firebase instance at the same time
 let alreadyStartedInitialization = false
```
Comment (theyokohamalife) [inline src/firebaseDb.ts:36]: Are you sure this is needed? I was under the impression that `initializeApp()` will throw an error if it's called more than once for the same application name. [source](https://stackoverflow.com/questions/75241561/where-does-firebase-getapps-length-check-come-from)

Comment (ermish) [reply src/firebaseDb.ts:36]: It was throwing an error during the tests since they run in parallel.

**Comment by ermish** [comment]
> @ermish Everything LGTM. I would just ask that we keep Husky for now, and I'm really sorry for the trouble! 🙇🏾‍♀️

ok @theyokohamalife  added back in husky 🐕

**Review by theyokohamalife — APPROVED** [review]
LGTM

**Pull request merged**